web/modules/custom/joinup_notification/joinup_notification.behat.inc

```diff
@@ -55,6 +55,7 @@ class JoinupNotificationSubContext extends DrupalSubContextBase implements Drupa
       throw new \Exception("User {$recipient} does not have an email.");
     }
 
+    \Drupal::state()->resetCache();
```

Comment:
This is how I think it works:
The behat process is a long running process, it starts at the start of the test, end ends at the end of it.
The actual sending of the mails happens somewhere in between the process. If however, a static gets populated in the start, and read back after other process set the state...